internal/common/backoff/retry.go

```diff
@@ -63,6 +65,16 @@ func (c *ConcurrentRetrier) throttleInternal(doneCh <-chan struct{}) time.Durati
 	c.Lock()
 	if c.failureCount > 0 {
 		next = c.retrier.NextBackOff()
+		// If secondary is included, use the greatest of the two (which also means
+		// if one is "done", which is -1, the one that's not done is chosen)
+		if c.includeSecondaryRetrier {
+			c.includeSecondaryRetrier = false
+			if c.secondaryRetrier != nil {
+				if secNext := c.secondaryRetrier.NextBackOff(); secNext > next {
+					next = secNext
+				}
+			}
+		}
```

Comment:
This turned out hard to parse IMHO

Reply:
Should the secondary retrier attempt be incremented even if it wasn't used? This would be consistent with the behavior defined in the issue and what I did in TS.

I don't think it's critical so I'm approving the PR.

Reply:
> Should the secondary retrier attempt be incremented even if it wasn't used?

I specifically asked at https://github.com/temporalio/sdk-features/issues/96#issuecomment-1201201389:

> Can you clarify the logic here when, after resource exhausted retry policy is set, what we do when we receive non-resource-exhausted error? Should we completely reset the retry policy? Same question if it's a non-resource-exhausted error and then a resource exhausted error.

And was pointed to https://github.com/temporalio/temporal/blob/3182dc2cde75e3e49fccb1de2905db5d30db58ab/common/backoff/retry.go#L204-L206 which I emulated here.

Reply:
My comment describes different logic.
As mentioned, I don't think any of this matters that much as long as the SDK backoffs "more" on resource exhausted.
Just wanted to keep the logic consistent between the SDKs.

Reply:
Your approach would require larger Go alterations and is in conflict with @yiminc's. If between y'all your approach is the one wanted for Go, we can apply it.